darts/models/forecasting/forecasting_model.py

```diff
@@ -444,105 +696,202 @@ def historical_forecasts(
+                        " where we can make a prediction or train the model. "
+                        "Please check the time indexes of the series and covariates."
+                    ),
+                    logger,
                 )
 
-            forecast = self._predict_wrapper(
-                n=forecast_horizon,
-                series=train,
-                past_covariates=past_covariates,
-                future_covariates=future_covariates,
-                num_samples=num_samples,
+            # prepare the start parameter -> pd.Timestamp
+            if start is not None:
+
+                historical_forecasts_time_index = drop_before_index(
+                    historical_forecasts_time_index,
+                    series_.get_timestamp_at_point(start),
+                )
+
+            else:
+                if (retrain is not False) or (not self._fit_called):
+
+                    if train_length:
+                        historical_forecasts_time_index = drop_before_index(
+                            historical_forecasts_time_index,
+                            historical_forecasts_time_index[0]
+                            + max(
+                                (train_length - self.training_sample_time_index_length),
+                                0,
+                            )
+                            * series_.freq,
+                        )
+                    # if not we start training right away, but with 2 minimum points, so we start
+                    # 1 time step after the first trainable point.
+                    # (sklearn check that there are at least 2 points in the training set and it seems
+                    # rather reasonable)
+                    else:
+                        historical_forecasts_time_index = drop_before_index(
+                            historical_forecasts_time_index,
+                            historical_forecasts_time_index[0] + 1 * series_.freq,
+                        )
+
+                    # if retrain is False, fit hasn't been called yet and train_length None,
+                    # it means that the entire backtesting will be based on a set of two training samples
+                    # at the first step, so we warn the user.
+                    if (
+                        (not self._fit_called)
+                        and (retrain is False)
+                        and (not train_length)
+                    ):
+                        logger.warning(
+                            "Model has not been fitted yet, `start` and train_length are not specified. "
+                            " The model is not retraining during the historical forecasts. Hence the "
+                            "the first and only training will be done on 2 samples which could lead to "
+                            "hazardous results."
+                        )
+
+            # build the prediction times in advance (to be able to use tqdm)
```

Comment:
Shall we also (maybe optionally) use tqdm at the outer-loop level? If you keep a list and don't move to an iterator, I would say it'll be very important for users to be able to track progress at the outer loop level.

We could have maybe two levels of iterations tracking when verbose=True for the two outer and inner loops (I think it should be possible).

Reply:
In the meantime I "fixed" this by having a tqdm iterator on the outer loop only if there are >1 series (and otherwise using tqdm in the inner loop).